File: 15595.cpp
#include <iostream>
#include <set>
#include <map>
#include <string>
#include <iomanip>

using namespace std;

int main(){
	int n;
	cin >> n;
	set<string> solved;
	map<string, int> wrong;
	for(int i = 0; i < n; ++i){
		string code, name, result, memory, time, language, length;
		cin >> code >> name >> result >> memory >> time >> language >> length;
		if(name == "megalusion" || solved.find(name) != solved.end()){
			continue;
		}
		if(result[0] == '4'){
			solved.insert(name);
			if(wrong.find(name) == wrong.end()){
				wrong[name] = 0;
			}
		}else{
			if(wrong.find(name) == wrong.end()){
				wrong[name] = 1;
			}else{
				wrong[name] += 1;
			}
		}
	}
	int numerator = solved.size();
	int denominator = numerator;
	for(string s : solved){
		denominator += wrong[s];
	}
	if(denominator == 0){
		cout << 0;
	}else{
		cout << setprecision(20) << numerator * 100.0 / denominator;
	}
	return 0;
}

File: 15595.py
def main():
    n = int(input())
    solved = set()
    wrong = {}
    for _ in range(n):
        info = input().split()
        username, result = info[1], info[2]
        if username == "megalusion" or username in solved:
            continue
        if result == "4":
            solved.add(username)
            if username not in wrong:
                wrong[username] = 0
        else:
            if username in wrong:
                wrong[username] += 1
            else:
                wrong[username] = 1
    numerator = len(solved)
    denominator = len(solved) + sum(wrong[username] for username in solved)
    if denominator == 0:
        print(0)
    else:
        print(numerator * 100 / denominator)


main()
